src/config/config.c

```diff
@@ -88,64 +89,25 @@ Include the automatically generated configuration data
 #include "config/config.auto.c"
 
 /***********************************************************************************************************************************
-Static data for the currently loaded configuration
+Data for the currently loaded configuration
 ***********************************************************************************************************************************/
-static struct ConfigStatic
-{
-    MemContext *memContext;                                         // Mem context for config data (child of top context)
-
-    // Generally set by the command parser but can also be set by during execute to change commands, i.e. backup -> expire
-    ConfigCommand command;                                          // Current command
-    ConfigCommandRole commandRole;                                  // Current command role
-
-    String *exe;                                                    // Location of the executable
-    bool help;                                                      // Was help requested for the command?
-    StringList *paramList;                                          // Parameters passed to the command (if any)
-
-    // Group options that are related together to allow valid and test checks across all options in the group
-    struct
-    {
-        bool valid;                                                 // Is option group valid for the current command?
-        unsigned int indexMax;                                      // Max index in option group
-        unsigned int value;                                         // Does option group index have any values?
-    } optionGroup[CFG_OPTION_GROUP_TOTAL];
-
-    // Map options names and indexes to option definitions
-    struct
-    {
-        bool valid:1;                                               // Is option valid for current command?
-        bool negate:1;                                              // Is the option negated?
-        bool reset:1;                                               // Is the option reset?
-        unsigned int source:2;                                      // Where the option came from, i.e. ConfigSource enum
-
-        Variant *value;                                             // Value
-        Variant *defaultValue;                                      // Default value
-    } option[CFG_OPTION_TOTAL];
-} configStatic;
+Config *configLocal = NULL;
 
 /**********************************************************************************************************************************/
 void
-cfgInit(void)
+cfgInit(Config *config)
```

Comment:
Shouldn't there be an ASSERT(config != NULL) in this function or is config allowed to be passed as a NULL (seems unlikely)?

Reply:
Yes, an assert would be good.

Reply:
fixed commit 9027d2753178917c15fbe8345751677534eeb628